### PR TITLE
Feature to fetch Auth Info from context when auth is enabled

### LINKED ISSUE #1094: Feature to fetch claims when authentication is enabled.
**Is your feature request related to a problem? Please describe.**
We can have a feature to get claims from GoFr context ....no matter the kind of authentication we use. 

**Describe the solution you'd like**
The usecase can be generalized for various kind of authentication like jwt or api key etc .....

### PR DESCRIPTION
**Pull request by coolwednesday**
**Description:**
-   Get claims/username/apikey through the context when auth is enabled.
-   Closes #1094 

**Checklist:**

-   [x] I have formatted my code using  `goimport`  and  `golangci-lint`.
-   [x] All new code is covered by unit tests.
-   [x] This PR does not decrease the overall code coverage.
-   [x] I have reviewed the code comments and documentation for clarity.

**Thank you for your contribution!**

### CODE REVIEW

File: pkg/gofr/context.go [hunk 1]
```diff
@@ -28,6 +31,12 @@ type Context struct {
 	responder Responder
 }
 
+type AuthInfo interface {
+	GetClaims() jwt.MapClaims
+	GetUsername() string
```
Comment (ccoVeille) [inline pkg/gofr/context.go:36]: I would suggest this "username" is term made of two words

```suggestion
	GetUserName() string
```

File: pkg/gofr/context.go [hunk 2]
```diff
@@ -75,6 +84,40 @@ func (c *Context) WriteMessageToSocket(data any) error {
...
+		return ""
+	}
+
+	return claims
+}
+
+func (c *Context) GetAPIKey() string {
+	claims, ok := c.Request.Context().Value(middleware.APIKey).(string)
+	if !ok {
+		c.Warn("failed to get api key from context")
```
Comment (ccoVeille) [inline pkg/gofr/context.go:114]:
```suggestion
		c.Warn("failed to get API key from context")
```

File: pkg/gofr/context.go [hunk 3]
```diff
@@ -75,6 +84,40 @@ func (c *Context) WriteMessageToSocket(data any) error {
...
+		return nil
+	}
+
+	return claims
+}
+
+func (c *Context) GetUsername() string {
+	claims, ok := c.Request.Context().Value(middleware.Username).(string)
+	if !ok {
+		c.Warn("failed to get username from context")
```
Comment (ccoVeille) [inline pkg/gofr/context.go:104]: I wouldn't use the failed wording on a non-blocking error.

You are using a warning error level, so I would use something like


```suggestion
		c.Warn("no username found in context")
```

Comment (ccoVeille) [reply pkg/gofr/context.go:104]: But I'm unsure about the logic of logging here

For me, there should be no log, it should be either:
- returning an error when missing
- returning simply a void result, like you do

In both cases, it should be from caller responsibility to handle this case

Imagine this log entry could be repeated fof ever and ever

File: pkg/gofr/context_test.go [hunk 4]
```diff
@@ -7,6 +7,9 @@ import (
 	"net/http/httptest"
 	"testing"
 
+	"github.com/golang-jwt/jwt/v5"
```
Comment (ccoVeille) [inline pkg/gofr/context_test.go:10]: Please fix this

You can use gci

File: pkg/gofr/http/middleware/basic_auth.go [hunk 5]
```diff
@@ -16,6 +17,8 @@ type BasicAuthProvider struct {
 	Container                   *container.Container
 }
 
+const Username authMethod = 1
```
Comment (ccoVeille) [inline pkg/gofr/http/middleware/basic_auth.go:20]:
```suggestion
const UserName authMethod = 1
```

**Review by ccoVeille — COMMENTED** [review]
Hello, it's me again 😅

File: pkg/gofr/context.go (hunk 1 above)
Comment (coolwednesday) [reply pkg/gofr/context.go:36]: I feel username is a single word , so it should not be UserName.

File: pkg/gofr/http/middleware/basic_auth.go (hunk 5 above)
Comment (coolwednesday) [reply pkg/gofr/http/middleware/basic_auth.go:20]: username is a single word.

File: pkg/gofr/context.go [hunk 6]
```diff
@@ -3,6 +3,9 @@ package gofr
 import (
 	"context"
 
+	"github.com/golang-jwt/jwt/v5"
+	"gofr.dev/pkg/gofr/http/middleware"
+
 	"github.com/gorilla/websocket"
```
Comment (ccoVeille) [inline pkg/gofr/context.go:9]:
```suggestion
	"gofr.dev/pkg/gofr/http/middleware"

	"github.com/golang-jwt/jwt/v5"
	"github.com/gorilla/websocket"
```

**Review by ccoVeille — APPROVED** [review]

**Review by Umang01-hash — APPROVED** [review]

File: pkg/gofr/context.go [hunk 7]
```diff
@@ -3,6 +3,9 @@ package gofr
 import (
 	"context"
 
+	"gofr.dev/pkg/gofr/http/middleware"
```
Comment (ccoVeille) [inline pkg/gofr/context.go:6]: Move it at the end with the other gofr.dev one

**Review by vipul-rawat — APPROVED** [review]

**Pull request merged**